app/javascript/app/phone-internationalization.js

```diff
@@ -10,58 +10,50 @@ const selectedInternationCodeOption = () => {
   return /** @type {HTMLOptionElement} */ (dropdown.item(dropdown.selectedIndex));
 };
 
-const unsupportedInternationalPhoneOTPDeliveryWarningMessage = () => {
-  const selectedOption = selectedInternationCodeOption();
-  if (selectedOption.dataset.smsOnly === 'true') {
-    const messageTemplate = I18n.t(
-      'two_factor_authentication.otp_delivery_preference.phone_unsupported',
-    );
-    return messageTemplate.replace('%{location}', selectedOption.dataset.countryName);
-  }
-  return null;
-};
+const setRadioEnabled = (radio, isEnabled) => {
+  radio.disabled = !isEnabled;
 
-const disablePhoneState = (
-  phoneRadio,
-  phoneLabel,
-  smsRadio,
-  deliveryMethodHint,
-  warningMessage,
-) => {
-  phoneRadio.disabled = true;
-  phoneLabel.classList.add('btn-disabled');
-  smsRadio.click();
-  deliveryMethodHint.innerText = warningMessage;
-};
+  const label = /** @type {Element} */ radio.parentNode.parentNode;
 
-const enablePhoneState = (phoneRadio, phoneLabel, deliveryMethodHint) => {
-  phoneRadio.disabled = false;
-  phoneLabel.classList.remove('btn-disabled');
-  deliveryMethodHint.innerText = I18n.t(
-    'two_factor_authentication.otp_delivery_preference.instruction',
-  );
+  if (isEnabled) {
+    label.classList.remove('btn-disabled');
+  } else {
+    label.classList.add('btn-disabled');
+  }
```

Comment:
A good candidate for [`toggle`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle):

```suggestion
  label.classList.toggle('btn-disabled', !isEnabled);
```

Reply:
oooo big fan, thanks!